### PR TITLE
floor and cell with precision

### PR DESCRIPTION
**Pull request by rnd-debug**
Provides `ceil(x,n)` and `floor(x,n)`. Replies to #1901.

- Shamelessly steals from https://github.com/sindresorhus/round-to as **round-to** seems to have a nice handling of rounding precision.
- Had to import **Decimal** in order to use `myBigNumber.toDecimalPlaces(n.toNumber(), Decimal.ROUND_CEIL)`

### CODE REVIEW

**Comment by josdejong** [comment]
Thanks @rnd-debug , looks good at first sight 👍 . I hope to review your PR coming weekend.

**Comment by rnd-debug** [comment]
One test case is failing on Firefox. Still haven't found the time to check what is happening. Hoping to do it this weekend.

**Comment by rnd-debug** [comment]
@josdejong failing tests fixed: ready to be merged.

**Comment by josdejong** [comment]
Your PR looks good @rnd-debug , thanks 👍 . Let's just ignore the codecov warnings, I'm not sure what's going on there but the new functionality is well tested.


A few feedback points:

- Can you add the new syntax `math.floor(x, n)` to the main comment in floor.js (used for the docs), like you did for `math.ceil(x, n)`?
- I didn't think about it before, but maybe we should also implement rounding with precision for the function `fix`, then all related functions (round, floor, ceil, fix) have the same behavior. What do you think?

**Comment by rnd-debug** [comment]
@josdejong fixing the 'fix' looks like a reasonable idea: will give it more time this weekend. Keeping in mind docs + trying to hack codecov by providing more tests.

**Comment by rnd-debug** [comment]
@josdejong I am confused about how to deal with rounding errors. According to the proposal for `ceil(x,n)`, we will get the following:
```
mathjs.floor(7.999999999999999, 2);                             // 8  But should it be 8?
mathjs.floor(-7.999999999999999, 2);                           //  -8 
```
When the external library gives us :
```
roundTo.down(7.999999999999999, 2);                        // 7.99   not cool
roundTo.down(-7.999999999999999, 2);                      // -8  
```
Similar issues can appear with `mathjs.floor`. Should we be compliant to the external library?

**Comment by josdejong** [comment]
> @josdejong I am confused about how to deal with rounding errors. 
>
> ...
>
> Similar issues can appear with `mathjs.ceil`. Should we be compliant to the external library?

If you calculate `ceil(0.1 + 0.2, 1)`, what whould you expect to get? `0.1 + 0.2` returns `0.30000000000000004`, and technically `ceil(0.30000000000000004, 1)` should return `0.4`. But that's not what we expect: we expect the result to be `0.3`, right?

To work around these issues with `equal`, `larger`, `smaller`, but also `ceil`, `fix`, `floor` in mathjs, we first apply a rounding to EPSILON (`1e-12` by default) to get rid of those round off errors. After that, we can do the actualy thing we want to do: checking equality, or applying ceil/fix/floor. Does that make sense?

**Comment by rnd-debug** [comment]
@josdejong Ok, now I got the big picture. Then the branch is ready to be merged.
Also provided a couple of tests for `round` that seemed to raise the coverage.

File: src/function/arithmetic/round.js [hunk 1]
```diff
@@ -34,10 +34,14 @@ export const createRound = /* #__PURE__ */ factory(name, dependencies, ({ typed,
...
    * Examples:
    *
-   *    math.round(3.2)              // returns number 3
-   *    math.round(3.8)              // returns number 4
+   *    math.round(3.22)             // returns number 3
+   *    math.round(3.82)             // returns number 4
    *    math.round(-4.2)             // returns number -4
    *    math.round(-4.7)             // returns number -5
+   *    math.round(3.22, 1)          // returns number 3.2
+   *    math.round(3.88, 1)          // returns number 3.8
```
Comment (josdejong) [inline src/function/arithmetic/round.js:42]: I think `math.round(3.88, 1)` returns `3.9` instead of `3.8`, right?

**Comment by josdejong** [comment]
Thanks for implementing this extension for `fix` too, and for the extensive unit tests 👍 ! All looks good, I'll fix the typo I saw in the comments of  `round`.

**Comment by josdejong** [comment]
Available now in `v7.4.0`

**Pull request merged**